core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregationBuilder.java

```diff
@@ -192,29 +192,33 @@ public TermsAggregationBuilder shardMinDocCount(long shardMinDocCount) {
     /**
      * Sets the order in which the buckets will be returned.
      */
-    public TermsAggregationBuilder order(Terms.Order order) {
+    public TermsAggregationBuilder order(BucketOrder order) {
         if (order == null) {
             throw new IllegalArgumentException("[order] must not be null: [" + name + "]");
         }
-        this.order = order;
+        if(order instanceof CompoundOrder || InternalOrder.isKeyOrder(order)) {
+            this.order = order;
+        } else {
+            this.order = BucketOrder.compound(order);
+        }
         return this;
     }
 
     /**
      * Sets the order in which the buckets will be returned.
      */
-    public TermsAggregationBuilder order(List<Terms.Order> orders) {
+    public TermsAggregationBuilder order(List<BucketOrder> orders) {
         if (orders == null) {
             throw new IllegalArgumentException("[orders] must not be null: [" + name + "]");
         }
-        order(Terms.Order.compound(orders));
+        order(orders.size() > 1 ? BucketOrder.compound(orders) : orders.get(0));
```

Comment:
This method has to handle a few different things

1. Handle a list of orders from parsed from xcontent (list or single order)
2. Add tie breaker `_key` order if needed
3. consistently convert the agg builder to/from xcontent

The old code had a bug with 3 if `order(BucketOrder order)` was called instead of `order(List<BucketOrder> orders)` because the later was used to parse the xcontent which resulted in a compound order instead of the individual order. This is now fixed and the test now calls both of these `order()` methods (randomly).